p4include/psa.p4

```diff
@@ -25,403 +24,518 @@ limitations under the License.
 }
 
 struct psa_ingress_input_metadata_t {
+  // All of these values are initialized by the architecture before
+  // the Ingress control block begins executing.
   PortId_t                 ingress_port;
-  InstanceType_t           instance_type;  /// Clone or Normal
-  /// set by the runtime in the parser, these are not under programmer control
-  ParserStatus_t           parser_status;
-  ParserErrorLocation_t    parser_error_location;
+  PacketPath_t             packet_path;
+  Timestamp_t              ingress_timestamp;
+  ParserError_t            parser_error;
 }
-
+// BEGIN:Metadata_ingress_output
 struct psa_ingress_output_metadata_t {
-  PortId_t                 egress_port;
+  // The comment after each field specifies its initial value when the
+  // Ingress control block begins executing.
+  ClassOfService_t         class_of_service; // 0
+  bool                     clone;            // false
+  PortId_t                 clone_port;       // undefined
+  ClassOfService_t         clone_class_of_service; // 0
+  bool                     drop;             // true
+  bool                     resubmit;         // false
+  MulticastGroup_t         multicast_group;  // 0
+  PortId_t                 egress_port;      // undefined
+  bool                     truncate;         // false
+  PacketLength_t           truncate_payload_bytes;  // undefined
 }
-
+// END:Metadata_ingress_output
 struct psa_egress_input_metadata_t {
+  ClassOfService_t         class_of_service;
   PortId_t                 egress_port;
-  InstanceType_t           instance_type;  /// Clone or Normal
+  PacketPath_t             packet_path;
   EgressInstance_t         instance;       /// instance coming from PRE
+  Timestamp_t              egress_timestamp;
+  ParserError_t            parser_error;
 }
-
-/* --------------------------------------------------------------------- */
-/**
- * Additional supported match_kind types
- */
+// BEGIN:Metadata_egress_output
+struct psa_egress_output_metadata_t {
+  // The comment after each field specifies its initial value when the
+  // Egress control block begins executing.
+  bool                     clone;         // false
+  ClassOfService_t         clone_class_of_service; // 0
+  bool                     recirculate;   // set by environment
+  bool                     drop;          // false
+  bool                     truncate;      // false
+  PacketLength_t           truncate_payload_bytes;  // undefined
+}
+// END:Metadata_egress_output
+// END:Metadata_types
+
+/// During the IngressDeparser execution, psa_clone_i2e returns true
+/// if and only if a clone of the ingress packet is being made to
+/// egress for the packet being processed.  If there are any
+/// assignments to the out parameter clone_i2e_meta in the
+/// IngressDeparser, they must be inside an if statement statement
+/// that only allows those assignments to execute if
+/// psa_clone_i2e(istd) returns true.  It can be implemented by
```

Comment:
I don't understand what 'It' refers to.

Reply:
I will update the version in p4-spec repository to say "psa_clone_i2e" instead of "It".

Reply:
Similarly for several other similar occurrences for "It can be implemented" phrases for other similar extern functions.